zathura/commands.c

```diff
@@ -149,6 +149,69 @@ bool cmd_bookmark_open(girara_session_t* session, girara_list_t* argument_list)
   return true;
 }
 
+bool cmd_jumplist_list(girara_session_t* session, girara_list_t* argument_list) {
+  g_return_val_if_fail(session != NULL, false);
+  g_return_val_if_fail(session->global.data != NULL, false);
+  zathura_t* zathura = session->global.data;
+  if (zathura->document == NULL) {
+    girara_notify(session, GIRARA_ERROR, _("No document opened."));
+    return false;
+  }
+
+  int num_entries = 5;
+
+  const unsigned int argc = girara_list_size(argument_list);
+  if (argc == 1) {
+    const char* value = girara_list_nth(argument_list, 0);
+    if (value != NULL) {
+      if (atoi(value) == 0) {
+        girara_notify(session, GIRARA_WARNING, _("Argument must be a nonzero number."));
+        return false;
+      }
+      if (atoi(value) > 0) {
+        num_entries = atoi(value);
+      } else {
+        num_entries = (int)zathura->jumplist.size + atoi(value);
+      }
+    }
+  } else if (argc != 0) {
+    girara_notify(session, GIRARA_ERROR, _("Invalid number of arguments given."));
+    return false;
+  }
+
+  if (num_entries <= 0) {
+    return true;
+  }
+
+  zathura_jump_t* current_jump = zathura_jumplist_current(zathura);
+  girara_list_iterator_t* iter = girara_list_iterator(zathura->jumplist.list);
+  while (girara_list_iterator_has_next(iter)) {
+    girara_list_iterator_next(iter);
+  }
```

Comment:
This loop and the next loop can be simplified by iterating from `size - 1` down.  The use of iterators should mostly go away.